File: BOJ/Silver IV/BOJ10828.py
#BOJ10828 스택 20210130
import sys
input = sys.stdin.readline

class Stack:
    def __init__(self):
        self.element = []
        self.stackSize = 0

    def push(self, x):
        self.element.append(x)
        self.stackSize += 1

    def pop(self):
        if self.element:
            print(self.element.pop())
            self.stackSize -= 1
        else:
            print(-1)

    def size(self):
        print(self.stackSize)

    def empty(self):
        print(0 if self.element else 1)

    def top(self):
        if self.element:
            print(self.element[-1])
        else:
            print(-1)

    def operate(self, args):
        if len(args) == 2:
            return{
                "push" : self.push
            }[args[0]](int(args[1]))
        else:
            return{
                "pop" : self.pop,
                "size" : self.size,
                "empty" : self.empty,
                "top" : self.top,
            }[args[0]]()
def main():
    s = Stack()
    for _ in range(int(input())):
        ins = input().rsplit()
        s.operate(ins)
if __name__ == '__main__':
    main()

File: BOJ/Silver IV/BOJ10828.java
//BOJ10828 스택 20210707
import java.io.*;
import java.util.*;

import static java.lang.Math.*;

public class Main {
    private static final BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    private static final BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    private static int n;

    public static void main(String[] args) throws IOException {
        n = Integer.parseInt(br.readLine());
        Stack<Integer> stack = new Stack<>();

        for(int i = 0; i < n; i++){
            String cmd = br.readLine();
            try {
                if (cmd.startsWith("push")) {
                    int v = Integer.parseInt(cmd.split(" ")[1]);
                    stack.push(v);
                } else if (cmd.equals("top")) {
                    bw.write(stack.peek() + "\n");
                } else if (cmd.equals("size")) {
                    bw.write(stack.size() + "\n");
                } else if (cmd.equals("empty")) {
                    bw.write(((stack.isEmpty()) ? 1 : 0) + "\n");
                } else if (cmd.equals("pop")) {
                    bw.write(stack.pop() + "\n");
                }
            }
            catch (Exception e){
                bw.write("-1\n");
            }
        }
        bw.flush();
        bw.close();
    }
}